### DISCUSSION TITLE
评论排序问题

### DISCUSSION THREAD

**Discussion by inkss** [ArtalkJS/Artalk#115] [Ideas]
层叠数设置成了2层，评论的排序能按照时间顺序来吗？现在似乎不是。

或者是层数+时间排序，相关地址：[复现地址](https://inkss.cn/blog/dc7bd0ff/)。

<details><summary>图示</summary><p>

![](https://user-images.githubusercontent.com/31947043/164274435-1b4a38fd-aaf4-4728-b702-df94833d6b98.png)

</p></details>

另外，Waline 没有转换器，哈哈哈哈哈。

**Comment by inkss** [comment]
有一个小小建议，地址中带 `#atk-comment-x` 的链接，跳转时能自动展开评论吗？被 *阅读更多* 隐藏了。

（如果这个好实现的话

**Reply by qwqcode** [reply]
自动展开已实现，排序问题也已修复，另外加上了回复 AT 显示

https://github.com/ArtalkJS/Artalk/commit/a5341c61dfe37bc985b01ee085d5a5fa11b76559

**Reply by qwqcode** [reply]
maxNesting 配置项名字改成了 nestMax

**Reply by inkss** [reply]
大赞👍

**Reply by qwqcode** [reply]
效果可见官网

**Reply by qwqcode** [reply]
https://artalk.js.org/#atk-comment-902

**Reply by inkss** [reply]
我觉得嵌套评论的排序，按照时间顺序让最新的在下面更符合阅读习惯。最新/最早只影响到他们的父评论呗

**Reply by inkss** [reply]
else if (sortBy === 'DATE_DESC') v = +new Date(a.comment.date) - +new Date(b.comment.date)

**Reply by qwqcode** [reply]
其实我也觉得，但是让最新的在上面能更方便找到评论

**Reply by qwqcode** [reply]
之后更新，学学 GitHub 
![image](https://user-images.githubusercontent.com/22412567/166139270-3f0edfe4-0e2d-49cd-8e51-23905b0a3466.png)

**Reply by inkss** [reply]
你这个评论系统最最最最大的亮点是有通知中心呀，那里面保持最新的在最上面就行了。

**Reply by qwqcode** [reply]
增加配置项 nestSort，可以自定义 https://github.com/ArtalkJS/Artalk/commit/91a0406d4451c810fc29f86d4c4514c0ffdbca64

**Reply by inkss** [reply]
大佬牛批                   ！！！！！！！！！！！（虽然我现在得再去合并、部署了 （溜~

**Reply by qwqcode** [reply]
刚刚修了个侧边栏bug，回复消息其实已经成功发出去了，但界面没有反应

后端也改了好多东西，可以升级下

[ArtalkGo Cache.pdf](https://github.com/ArtalkJS/Artalk/files/8598646/ArtalkGo.Cache.pdf)

**Reply by qwqcode** [reply]
前端代码太乱了，因为开发了几年（

也懒得整理  😅😅😅

**Reply by inkss** [reply]
先优化导入，格式化一波 (╯▔皿▔)╯

![Artalk – list-lite ts_22-05-01_17-48-56](https://user-images.githubusercontent.com/31947043/166140771-8473086b-1f0b-471a-96f5-93502cee9be9.png)

**Reply by inkss** [reply]
```bash
➜  artalk ./artalk-go upgrade   
INFO[0000] 从 GitHub Release 检索更新中...                    
INFO[0001] 当前已是最新版本 v2.1.4 无需升级   
```

**Reply by xiamuguizhi** [reply]
@inkss Artalk (v2.1.5/5d2c090) 看到艾特功能了。 哈哈。

**Reply by qwqcode** [reply]
刚刚发布了，后端增加支持分配站点管理员

https://artalk.js.org/guide/backend/config.html#admin-users-%E7%AE%A1%E7%90%86%E5%91%98%E8%B4%A6%E6%88%B7

**Reply by qwqcode** [reply]
你可以和你的朋友们共用同一个 ArtalkGo

**Reply by xiamuguizhi** [reply]
@qwqcode 我不添加到conf.yml 应该不冲突吧，我一个人用挺好的 哈哈哈哈！！

**Reply by inkss** [reply]
> 刚刚发布了，后端增加支持分配站点管理员
> 
> https://artalk.js.org/guide/backend/config.html#admin-users-%E7%AE%A1%E7%90%86%E5%91%98%E8%B4%A6%E6%88%B7

这个功能好哇，以前最多只能算是一人多站点

**Reply by qwqcode** [reply]
v2.1.5 升级无须修改 conf.yml，向下兼容的

**Comment by kecrily** [comment]
如果 Waline 用的是 LeanCloud 的话，Valine 的转换器也是适用的。See https://artalk.js.org/guide/transfer.html#valine-waline

如果是用的 MySQL/PostgreSQL 之类的话，兼容成本就有点高了。Waline 支持的数据库实在是太多了

**Reply by inkss** [reply]
那还是让杜老师手撸SQL吧，哈哈哈，他觉得waline提交评论得等两三秒，被我安利这个了

**Reply by xiamuguizhi** [reply]
@inkss 哈哈 只能辛苦杜老师了， 估计杜老师就几个命令就好了，不像我 当初复制评论数据库 用了6小时！！真一条条复制。

**Reply by penndu** [reply]
我也弄了几个小时，实在没耐心了~

**Reply by xiamuguizhi** [reply]
@penndu waline能不能转typecho 在从 typecho 转过来

**Reply by qwqcode** [reply]
@penndu waline能导出json格式数据吗

**Reply by penndu** [reply]
> @penndu waline能不能转typecho 在从 typecho 转过来

好像不行~

**Reply by penndu** [reply]
> @penndu waline能导出json格式数据吗

我没有json数据，都是mysql的~

**Reply by qwqcode** [reply]
@penndu 那这周末我有空看下做吧

**Reply by qwqcode** [reply]
@penndu waline 都不能导出 json，太捞了吧

**Reply by penndu** [reply]
> @penndu 那这周末我有空看下做吧

非常感谢！

**Reply by penndu** [reply]
> @penndu waline 都不能导出 json，太捞了吧

后端只有管理功能，没有备份导出~

**Reply by inkss** [reply]
你不能用其他工具连上数据库，导出SQL嘛？

**Reply by penndu** [reply]
> 你不能用其他工具连上数据库，导出SQL嘛？

![image](https://user-images.githubusercontent.com/46226528/164890672-040a3567-9403-45a1-b24b-7d4fedfa192b.png)

**Reply by inkss** [reply]
我有个想法，你把 waline 的和 artalk 的表先建在一个数据库里，然后 insert into select 呗，起码现在 artalk 表没关联，可以随便放数据。

**Reply by inkss** [reply]
```sql
insert into comments(created_at, updated_at, content, page_key, site_name, user_id, ua, ip)
select createdAt, updateAt, comment, url, '你的网站名字', user_id, ua, ip from wl_comment;
```

**Reply by penndu** [reply]
> 我有个想法，你把 waline 的和 artalk 的表先建在一个数据库里，然后 insert into select 呗，起码现在 artalk 表没关联，可以随便放数据。

评论关联就没有了~

**Reply by inkss** [reply]
另外需要处理的表是 pages, users，sites 手动加一条就行了。

**Reply by inkss** [reply]
关联是 rid，把这个照抄吧，不过这样前面的 id 也得加进去了

**Reply by inkss** [reply]
主要是表都在，插数据就简单了呀，各种查呗

**Reply by qwqcode** [reply]
Artransfer-CLI 工具发布，支持从 Waline 数据库导出 Artrans：https://github.com/ArtalkJS/Artransfer-CLI/releases

**Reply by qwqcode** [reply]
![image](https://user-images.githubusercontent.com/22412567/164979827-9cc82384-53c9-436e-82f0-2457d485e6bb.png)

**Reply by penndu** [reply]
> ![image](https://user-images.githubusercontent.com/22412567/164979827-9cc82384-53c9-436e-82f0-2457d485e6bb.png)

今晚上不睡了，就搞这个！
